### PR TITLE
ZCS-11830: Added accountCosDomainInherited flag for zimbraDumpsterEna…

### PR DESCRIPTION
**Pull request by zimsuchitgupta**
Added accountCosDomainInherited flag for zimbraDumpsterEnabled attribute

### CODE REVIEW

**Review by dasiyogesh — APPROVED** [review]

File: store/conf/attrs/zimbra-attrs.xml [hunk 1]
```diff
@@ -5260,7 +5260,7 @@ TODO: delete them permanently from here
   <desc>enable/disable MAPI (Microsoft Outlook) Connector</desc>
 </attr>
 
-<attr id="1128" name="zimbraDumpsterEnabled" type="boolean" cardinality="single" optionalIn="globalConfig,account,cos,domain" flags="accountInfo,accountInherited,domainAdminModifiable,domainInherited" since="7.0.0">
+<attr id="1128" name="zimbraDumpsterEnabled" type="boolean" cardinality="single" optionalIn="globalConfig,account,cos,domain" flags="accountInfo,accountCosDomainInherited,domainAdminModifiable,domainInherited" since="7.0.0">
```
Comment (deepakzimb) [inline store/conf/attrs/zimbra-attrs.xml:5263]: No change in associated files? Like we run get-set steps to generate definitions?

Comment (silentsakky) [inline store/conf/attrs/zimbra-attrs.xml:5263]: check how this works in upgrade scenario

Comment (zimsuchitgupta) [reply store/conf/attrs/zimbra-attrs.xml:5263]: Had run `ant generate-getters` found no diff.

**Review by lavegupta — APPROVED** [review]

**Review by dasiyogesh — APPROVED** [review]

File: store/conf/attrs/zimbra-attrs.xml (hunk 1 above)
Comment (zimsuchitgupta) [reply store/conf/attrs/zimbra-attrs.xml:5263]: We don't require any extra work to do, this attribute will be reflected while upgradation.

**Review by silentsakky — APPROVED** [review]

**Review by deepakzimb — APPROVED** [review]

File: store/conf/attrs/zimbra-attrs.xml (hunk 1 above)
Comment (k-kato) [reply store/conf/attrs/zimbra-attrs.xml:5263]: Will new `<globalConfigValue>FALSE</globalConfigValue>` , which is added on https://github.com/Zimbra/zm-mailbox/pull/1314, be applied on an existing attribute at upgrade?

**Comment by zimsuchitgupta** [comment]
> I think it is related to #1314 How about the following attributes? Don't we need to update them in the same way?
> 
> ```
> zimbraMailDumpsterLifetime
> zimbraDumpsterUserVisibleAge
> zimbraDumpsterPurgeEnabled
> ```

Thanks @k-kato , updated the flag in these attributes too.

**Comment by silentsakky** [comment]
> > I think it is related to #1314 How about the following attributes? Don't we need to update them in the same way?
> > ```
> > zimbraMailDumpsterLifetime
> > zimbraDumpsterUserVisibleAge
> > zimbraDumpsterPurgeEnabled
> > ```
> 
> Thanks @k-kato , updated the flag in these attributes too.

Thanks @zimsuchitgupta also mention these changes in ticket so QA team is aware and can cover additional scope

File: store/conf/attrs/zimbra-attrs.xml [hunk 2]
```diff
@@ -6603,13 +6603,13 @@ TODO: delete them permanently from here
...
 
-<attr id="1314" name="zimbraDumpsterUserVisibleAge" type="duration" cardinality="single" optionalIn="globalConfig,domain,account,cos" flags="accountInherited,domainInherited,domainAdminModifiable" since="8.0.0">
+<attr id="1314" name="zimbraDumpsterUserVisibleAge" type="duration" cardinality="single" optionalIn="globalConfig,domain,account,cos" flags="accountCosDomainInherited,domainInherited,domainAdminModifiable" since="8.0.0">
   <globalConfigValue>30d</globalConfigValue>
   <defaultCOSValue>30d</defaultCOSValue>
   <desc>limits how much of a dumpster data is viewable by the end user, based on the age since being put in dumpster</desc>
 </attr>
 
-<attr id="1315" name="zimbraDumpsterPurgeEnabled" type="boolean" cardinality="single" optionalIn="globalConfig,domain,account,cos" flags="accountInherited,domainInherited,domainAdminModifiable" since="8.0.0">
+<attr id="1315" name="zimbraDumpsterPurgeEnabled" type="boolean" cardinality="single" optionalIn="globalConfig,domain,account,cos" flags="accountCosDomainInherited,domainInherited,domainAdminModifiable" since="8.0.0">
```
Comment (silentsakky) [inline store/conf/attrs/zimbra-attrs.xml:6612]: @lavegupta do you think this will have any impact on ui functionality?

**Review by k-kato — APPROVED** [review]

File: store/conf/attrs/zimbra-attrs.xml (hunk 2 above)
Comment (lavegupta) [reply store/conf/attrs/zimbra-attrs.xml:6612]: No, It will not.

File: store/conf/attrs/zimbra-attrs.xml (hunk 1 above)
Comment (zimsuchitgupta) [reply store/conf/attrs/zimbra-attrs.xml:5263]: @k-kato Verified this on after upgrading the build, we are seeing the expected value :
`zimbra@perf-zcs-4-zimbradev-com:~$ zmprov gcf zimbraDumpsterEnabled`
`zimbraDumpsterEnabled: TRUE`
`zimbra@perf-zcs-4-zimbradev-com:~$ zmprov gcf zimbraDumpsterUserVisibleAge`
`zimbraDumpsterUserVisibleAge: 30d`
`zimbra@perf-zcs-4-zimbradev-com:~$ zmprov gcf zimbraDumpsterPurgeEnabled`
`zimbraDumpsterPurgeEnabled: TRUE`

**Comment by zimsuchitgupta** [comment]
> zimbraMailDumpsterLifetime
> zimbraDumpsterUserVisibleAge
> zimbraDumpsterPurgeEnabled

Added @silentsakky

File: store/conf/attrs/zimbra-attrs.xml (hunk 1 above)
Comment (k-kato) [reply store/conf/attrs/zimbra-attrs.xml:5263]: @zimsuchitgupta thank you for the confirmation.

**Pull request merged**